### ISSUE TITLE
Base data export on latest published survey version

### ISSUE THREAD

**Issue by janagombitova** [akvo/akvo-flow#2285]
# Context
Based on the research and discussions noted here: akvo/akvo-product-design#227 we have agreed to make multiple changes around exports.

Currently when the users export data we take the current form structure (published or not published). What we propose is to base the exported data structure on the last published version of the survey form. This is because the data is collected against published versions. There are cases when Salim changes the form but data is still collected thus the data export should be based on the most likely version of the form the data is collected to (last published form version). 

We need to, however, make sure that this does not affect the Survey form export. Sometimes users use it to cross check the survey structure before publishing and sending it out for data collection. 

# Questions 
- What will happen if the user creates form for the 1st time and then tries to do the data export?

**Comment by stellanl** [comment]
Since the Survey form export is done by different code, keeping them the same will be trivial.
The report engine will need to fetch and parse a published survey file instead of reading the data store. We could take this opportunity to keep all published versions around by adding a version number to the filename in S3, or keep it like today where the versionless file in S3 is implicitly the latest one published.

**Comment by stellanl** [comment]
There is a more ambitious approach to this. We could introduce version management to Flow. Think of the publishing process as a software release or a Git commit.

When you publish a survey, we save a snapshot of the entire survey at that time, *in the regular db*. All question groups, questions and options get a version field that ties them together. (We probably also want a data entity for the publication/release itself, to remember when, who etc..)

**Comment by stellanl** [comment]
There is a bug with NUMBER answers. Old code makes "1.0" into 1 and saves it to the spreadsheet as a number - new code saves it as the string "1.0"

**Comment by janagombitova** [comment]
I created a survey form and published it and submitted some data to it. Then exported the data. After that I changed the form by adding a new group but I did not publish it. I exported the survey form and saw my changes - as expected. I exported the Data cleaning export and the file structure was based on the NOT published version. So it seems these changes are either not on UAT1, or there is a bug, or am I missing something?  @stellanl
